src/main/java/com/github/mmolimar/kafka/connect/fs/FsSourceTask.java

```diff
@@ -116,15 +143,14 @@ public void start(Map<String, String> properties) {
         }
     }
 
-    private <T> Stream<T> asStream(Iterator<T> src) {
-        Iterable<T> iterable = () -> src;
-        return StreamSupport.stream(iterable.spliterator(), false);
-    }
-
     private SourceRecord convert(FileMetadata metadata, long offset, Struct struct) {
+        Map<String, Long> offsetMap = new HashMap<>();
+        offsetMap.put("offset", offset);
+        offsetMap.put("fileSizeBytes", metadata.getLen());
```

Comment:
If you store the whole file length here, you risk skipping messages. Specifically in the case where only a subset of the messages in a file have been committed.
Maybe it would probably be better to store the amount of bytes read thus far?

Reply:
Right now we either process an entire file and commit the offsets or we don't process the file at all right? Is there a situation where we could process part of a file? I didn't think that was possible in the current code.

Reply:
Didn't think about that behaviour in the exception handling , maybe it would be better to not send any records for a file if any record fails then.